src/android/toga_android/window.py

```diff
@@ -26,17 +26,10 @@ def create(self):
         pass
 
     def set_app(self, app):
-        self._create()
+        pass
 
     def set_content(self, widget):
-        if self.native is None:
-            widget.native = TogaLayout(self.app.native, widget)
-
-        # Add all children to the content widget.
-        for child in widget.interface.children:
-            child._impl.container = widget
-
-        self.app._impl.setContentView(self._container._impl)
+        MainActivity.singletonThis.setContentView(widget.native)
```

Comment:
Continuing the earlier thought about using set_app(); this could be `self.app.native.setContentView()`

Reply:
This one I agree with :) Sorry that it must look I'm being very stubborn.

I've pushed a fix for this within this PR.